fili-core/src/main/java/com/yahoo/bard/webservice/table/BaseCompositePhysicalTable.java

```diff
@@ -0,0 +1,90 @@
+
+    /**
+     * Constructor.
+     *
+     * @param name  Name that represents set of fact table names that are put together under this table
+     * @param columns  The columns for this table
+     * @param physicalTables  A set of PhysicalTables that are put together under this table. The
+     * tables will be used to compute common/coarsest time grain among them. The PhysicalTables needs to have mutually
+     * satisfying time grains in order to calculate the common/coarsest time grain.
+     * @param logicalToPhysicalColumnNames  Mappings from logical to physical names
+     * @param availability  The Availability of this table
+     */
+    public BaseCompositePhysicalTable(
+            @NotNull TableName name,
+            @NotNull Set<Column> columns,
+            @NotNull Set<PhysicalTable> physicalTables,
+            @NotNull Map<String, String> logicalToPhysicalColumnNames,
+            @NotNull Availability availability
+    ) {
+        super(
+                name,
+                IntervalUtils.getCoarsestTimeGrain(physicalTables).orElseThrow(() -> {
+                    String message = String.format(
+                            "At least 1 physical table needs to be provided in order to calculate " +
+                                    "coarsest time grain for %s",
+                            name.asName()
+                    );
+                    LOG.error(message);
+                    return new IllegalArgumentException(message);
+                }),
+                columns,
+                logicalToPhysicalColumnNames,
+                availability
+        );
+        verifyGrainSatisfiesAllTables(getSchema().getTimeGrain(), physicalTables, name);
+    }
+
+    /**
+     * Verifies that the coarsest ZonedTimeGrain satisfies all tables.
+     *
+     * @param coarsestTimeGrain  The coarsest ZonedTimeGrain to be verified
+     * @param physicalTables  A set of PhysicalTables whose ZonedTimeGrains are checked to make sure
+     * they all satisfies with the given coarsest ZonedTimeGrain
+     * @param name  Name of the current MetricUnionCompositeTable that represents set of fact table names
+     * joined together
+     *
+     * @throws IllegalArgumentException when there is no mutually satisfying grain among the table's time grains
+     */
+    private static void verifyGrainSatisfiesAllTables(
+            ZonedTimeGrain coarsestTimeGrain,
+            Set<PhysicalTable> physicalTables,
+            TableName name
+    ) throws IllegalArgumentException {
+        if (!physicalTables.stream()
+                .map(PhysicalTable::getSchema)
+                .map(PhysicalTableSchema::getTimeGrain)
+                .allMatch(grain -> grain.satisfiedBy(coarsestTimeGrain))) {
+            String message = String.format("There is no mutually satisfying grain among: %s for current table " +
+                            "%s",
```

Comment:
When wrapping, chop-wrap all parameters. That also means that the string doesn't need to be split either:

```java
String message = String.format(
        "There is no mutually satisfying grain among: %s for current table %s",
        physicalTables,
        name.asName()
);
```